src/upgrade/upgrade_scripts/5.10.0/upgrade_bucket_lifecycle_structure.js

```diff
@@ -0,0 +1,52 @@
+/* Copyright (C) 2016 NooBaa */
+"use strict";
+
+const util = require('util');
+
+async function run({ dbg, system_store, system_server }) {
+
+    try {
+        dbg.log0('starting upgrade lifecycle...');
+        let should_make_change = false;
+        const buckets = [];
+        for (const bucket of system_store.data.buckets) {
+            //Do not update if there are no  lifecycle_configuration_rules or if the bucket is being deleted.
+            if (!bucket.lifecycle_configuration_rules || bucket.deleting) return;
+
+            let iteratee = 0;
+            for (const lifecycle_rule of bucket.lifecycle_configuration_rules) {
+                if (lifecycle_rule.prefix) {
+                    dbg.log0('bucket.lifecycle_configuration_rules[', iteratee, ']', lifecycle_rule);
+                    lifecycle_rule.filter = {};
+                    lifecycle_rule.filter.prefix = lifecycle_rule.prefix;
+                    delete lifecycle_rule.prefix;
+                    bucket.lifecycle_configuration_rules[iteratee] = lifecycle_rule;
```

Comment:
I'm not sure why you need to do it inline - I think it's better to just create a new array of changes and push when needed, without changing the system_store (at least not before the make_changes)